core/blockchain.go

```diff
@@ -3158,6 +3219,147 @@ func (bc *BlockChain) IsEnablePruneBeaconChainFeature() bool {
 	return bc.pruneBeaconChainEnable
 }
 
+// SyncFromTiKVWriter used for tikv mode, all reader or follower writer used to sync block from master writer
+func (bc *BlockChain) SyncFromTiKVWriter(newBlkNum uint64, logs []*types.Log) error {
+	head := rawdb.ReadHeadBlockHash(bc.db)
+	dbBlock := bc.GetBlockByHash(head)
+	currentBlock := bc.CurrentBlock()
+
+	if dbBlock == nil || currentBlock == nil {
+		return nil
+	}
+
+	currentBlockNum := currentBlock.NumberU64()
+	if currentBlockNum < newBlkNum {
+		start := time.Now()
+		for i := currentBlockNum; i <= newBlkNum; i++ {
+			blk := bc.GetBlockByNumber(i)
+			if blk == nil {
+				// todo check later
```

Comment:
?? fixed or not? please leave a better comment here.

Reply:
optimized